### PR TITLE
Implement basic Discord bot

### PR DESCRIPTION
**Pull request by chompaa**
Adds a basic Discord bot implementation. 
- Bot token is set with the `DISCORD_TOKEN` environment variable.
- Run with `cargo run` in the `bot` directory.
- Currently has a slash command `ask` (only works in DMs) which repeats the user's query as output.

### CODE REVIEW

File: bot/cargo/config.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[env]
```
Comment (etuckerman) [inline bot/cargo/config.toml]: The parent folder should be `.cargo` not `cargo`.

Comment (chompaa) [reply bot/cargo/config.toml]: Good catch, thanks!

Comment (etuckerman) [reply bot/cargo/config.toml]: No problem!

**Review by etuckerman — APPROVED** [review]
Look good to me, thanks for doing this 😎👍

**Pull request merged**